### ISSUE TITLE
KerasModel.compile() should support sample_weight_mode

### ISSUE THREAD

**Issue by bzamecnik** [tensorpack/tensorpack#764]
Feature Request.

`keras.models.Model.compile(self, optimizer, loss=None, metrics=None, loss_weights=None, sample_weight_mode=None, weighted_metrics=None, target_tensors=None)`

while:

`tensorpack.contrib.keras.compile(self, optimizer, loss, metrics=None)`

Related to #763

**Comment by ppwwyyxx** [comment]
I don't think I'll be able to spend any more time on supporting Keras.
Plus I think having `sample_weight` is  a very ugly design..

**Comment by bzamecnik** [comment]
I don't say you have to do it. It's just a reminder for me to write it in
future if needed. We're still considering whether to use just DataFlow or
also trainers in our project since Keras support is still quite
experimental.

Dne út, 15. 5. 2018 18:53 uživatel Yuxin Wu <notifications@github.com>
napsal:

**Comment by ppwwyyxx** [comment]
Yes. Keras support is just a proof-of-concept. And obviously it is unlikely to be a focus of this project.

**Comment by bzamecnik** [comment]
Well, thanks for the awesome work. At Rossum we'll certainly need to
improve our research infrastructure to be both easy to use and high
performance. So far Tensorpack DataFlow seems to be a really good step -
well composable and faster than our home grown multi-processing generators.
On the other hand leaving all the Keras tools like model saving/loading,
callbacks and other stuff may be just a too big step for less marginal
benefit. Let's see if we'll be able to improve Tensorpack to support Keras
better, or improve Keras for TF Queue/Staging/Multi GPU or just switch to
Tensorflow completely. Anyway if there's any improvement we'll be happy to
provide PR's.

Dne út, 15. 5. 2018 22:08 uživatel Yuxin Wu <notifications@github.com>
napsal:
